sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRelTest.java

```diff
@@ -212,6 +217,20 @@ public void testOrderBy_nullsLast() throws Exception {
     pipeline.run().waitUntilFinish();
   }
 
+  @Test
+  public void testOrderBy_with_offset2() throws Exception {
+    Schema schema = Schema.builder().addField("count_star", Schema.FieldType.INT64).build();
+
+    String sql =
+        "INSERT INTO COUNT_TABLE(count_star) "
+            + "SELECT COUNT(*) FROM (SELECT * FROM "
+            + "(SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3) LIMIT 3 OFFSET 1)";
```

Comment:
If there is no ORDER BY, then the particular elements that show up in the LIMIT / OFFSET are not deterministic, right?

Reply:
Yes